irohad/network/peer_communication_stub.cpp

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2017 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <network/peer_communication_stub.hpp>
+
+namespace iroha {
+  namespace network {
+
+    using ametsuchi::Ametsuchi;
+    using validation::StatefulValidator;
+    using validation::ChainValidator;
+    using ordering::OrderingService;
+    using consensus::ConsensusService;
+    using dao::Block;
+
+    rxcpp::observable<rxcpp::observable<dao::Block>>
+    PeerCommunicationServiceStub::on_commit() {
+      return consensus_.on_commit().take_while([this](auto commit) {
+        auto storage = storage_.createMutableStorage();
+        auto result = chain_validator_.validate(commit, *storage);
+        if (result) {
+          storage_.commit(*storage);
+        }
+        return result;
+      });
+    }
+
+    void PeerCommunicationServiceStub::propagate_transaction(
+        dao::Transaction &tx) {
+      orderer_.propagate_transaction(tx);
+    }
+
+    rxcpp::observable<dao::Proposal>
+    PeerCommunicationServiceStub::on_proposal() {
+      return orderer_.on_proposal().tap([this](dao::Proposal &proposal) {
+        auto wsv = storage_.createTemporaryWsv();
+        auto validated_proposal = stateful_validator_.validate(proposal, *wsv);
```

Comment:
For Sumeragi and another consensus algorithms please look https://github.com/hyperledger/iroha/pull/415. We will have pluggable consensus, meaning you can use any consensus implementation in future. The only requirement for a consensus is that it must implement the interface ConsensusService and OrderingService.